azurelinuxagent/ga/update.py

```diff
@@ -937,31 +933,17 @@ def _download(self):
         uris_shuffled = self.pkg.uris
         random.shuffle(uris_shuffled)
         for uri in uris_shuffled:
-            if not HostPluginProtocol.is_default_channel() and self._fetch(uri.uri): # pylint: disable=R1723
```

Comment:
Why not move this function to use the `send_request_using_appropriate_channel` function too?

Reply:
`self` here is the `GuestAgent` object which does not have the reference to `WireClient` where the method `send_request_using_appropriate_channel` lives.

Reply:
Even the `_fetch` method called here is a private method of `GuestAgent`, not the `WireClient` (like other methods that fetch things).